DF-2391 - Allow codebuild job to invoke lambda in some environments

bugs404 opened the pull request:
- Add an optional variable that defaults to false and if true gives the codebuild job for the lambda function the ability to invoke the lambda function (for integration testing purposes)

Commit:
DF-2391

- Add an optional variable that defaults to false and if true gives the codebuild job for the lambda function the ability to invoke the lambda function (for integration testing purposes)

bugs404 commented on lambda_function/ci.tf:
```diff
@@ -49,6 +49,7 @@ resource "aws_iam_role_policy" "codebuild" {
         "lambda:UpdateFunctionCode",
         "lambda:ListVersionsByFunction",
         "lambda:UpdateAlias"
+        "${var.codebuild_can_invoke_lambda ? ",lambda:InvokeFunction" : ""}"
```
not sure if this is possible or I might have to break this out into separate statements and use the dynamic functionality.

bugs404 replied:
Testing in sandbox currently

Commit:
DF-2391

- Change the approach to having a dynamic iam policy for codebuild

Commit:
DF-2391

- rename variable that is the switch for running integration tests within the codebuild job for a lambda
- propagate the switch as an env variable available to the codebuild job so it can decide to run an integration test or not

Commit:
DF-2391

- fix environment variable declaration for codebuild job

Commit:
DF-2391

- add permission to get lambda function config so that when codebuild is configured to run ci against lambda it can check to make sure lambda is updated

eyablonowitz reviewed (changes requested):
Please document this new option in the README and include an example of how it could be used.

Commit:
DF-2391

- Update readme to document new functionality and environment variable that exposes said functionality

Commit:
DF-2391

- Add little excerpt at the end of readme to further describe how the variable could be used

Commit:
DF-2391

- small README fix

Commit:
DF-2391

- small README fix

eyablonowitz commented on README.md:
```diff
@@ -123,3 +124,11 @@ Ensure you add the following permissions to the lambda role
+ * The codebuild job that accompanies lambda ci is now able to invoke the lambda function
+ * The codebuild job will know if it's appropriate to perform integration testing in the environment it's running in according to env variable "run_integration_test"
+
+For an example implementation of a lambda-codebuild job setup to conditionally run integration tests see https://github.com/PatientPing/d2p_address_lookup_lambda
```
This is a public repo/module.  We don't want to put links to private repos in it.  But you could just copy https://github.com/PatientPing/d2p_address_lookup_lambda/blob/DF-2391/buildspec.yml#L16-L19 here to at least hint at how to wire this up.

Commit:
DF-2391

- Instead of linking to another private repo for an example implementation of a buildspec conditionally running an integration test, include just the salient buildspec excerpt in the README

eyablonowitz reviewed (approved):
lgtm
